.github/workflows/golden-test.yml

```diff
@@ -9,8 +9,16 @@ on:
 
 jobs:
   golden-test:
-    name: Golden Test
+    name: Golden Test ${{ matrix.grouping }}
     runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        include:
+          - grouping: 'w/ grouping'
+            group_errors: 'true'
+          - grouping: 'w/o grouping'
+            group_errors: 'false'
+
```

Comment:
Creating two jobs (that are running in parallel) is a good idea since golden test can easily run for several minutes.

But this would actually create two comments in the PR, which may hinder the readability for other comments.

Instead, can we keep the two jobs here (with id `golden-test-w-grouping` and `golden-test-wo-grouping`) [with output](https://docs.github.com/en/actions/using-jobs/defining-outputs-for-jobs), and extract the uploading step to be a separate job that depends on these two jobs, reads the output from these two jobs, and construct the message together and upload only once.

It shouldn't be too complex I think, and offers better messages for readability. What do you think?